### ISSUE TITLE
Heisenbug with graph.clj

### ISSUE THREAD

**Issue by malcolmsparks** [ztellman/byte-streams#25]
Have been getting the following error, which I can't reproduce on other machines.

I'm noting here just in case others are seeing the same thing.

I have no idea what is causing it, it's caused me a lot of pain this week. Just can't see what the fix is.

```
CompilerException java.lang.RuntimeException: No such var: s/->source, compiling:(byte_streams/graph.clj:108:64)
```

**Comment by ztellman** [comment]
What version?

**Comment by malcolmsparks** [comment]
0.2.2 - but I'm happy closing the issue, I'm convinced I've got some kind
of low-level or hardware issue that's to blame

**Comment by ztellman** [comment]
Is there any AOT compilation going on?

**Comment by ztellman** [comment]
There's some conditional compilation stuff going on in Manifold, at
https://github.com/ztellman/manifold/blob/master/src/manifold/deferred.clj#L1205
 and
https://github.com/ztellman/manifold/blob/master/src/manifold/stream.clj#L45,
which could throw an exception, which would cause the error you're seeing
(it "loads" the namespace, but it's empty because it throws while
loading).  My guess is something weird there, possibly because it's
compiled with JDK 8 and run under JDK 7?  Just a guess.

**Comment by malcolmsparks** [comment]
I've seen the conditional compilation in manifold and I guess it could be related. I have tried with and without core.async in the classpath.

However, this issue only crops up 60% of the time, the other times I start a REPL it is either fine, or other unexpected compilation errors are produced, and they are rarely the same. The only error that seems to come up quite frequently is this `No such var: s/->source` one.

I've ensured there is no AOT compilation going on, and that the git repo is clean with `git clean -fdx`. I've upgraded my Leiningen to 2.7.1, deleted my boot cache, deleted my `$HOME/.m2/repository` entirely (twice!), upgraded my machine (Arch, so latest openjdk 8). There's no java 7 remaining on my machine.

I've run out of ideas what could be causing this. Since it doesn't happen reliably, it's hard to reduce down to a reliably failing test, in fact, when I do so, the issue tends to disappear. I've had the issue with both Clojure 1.8 and Clojure 1.9-alpha14, so it's not related to Clojure version either. I can't think what else to try.

**Comment by ztellman** [comment]
Hmm.  Well, I'll keep an eye out for it, and let you know if I figure anything out.  Leaving this open for now.

**Comment by malcolmsparks** [comment]
I've finally managed to reduce the problem down to https://github.com/clojure-emacs/refactor-nrepl

There's an issue here: https://github.com/clojure-emacs/refactor-nrepl/issues/160 - I've determined that the fix mentions doesn't help me, but looking at what refactor-nrepl is doing leads me to think that there's something here that affects the ordering of requiring namespaces which triggers my problem.

Whatever it is, if I avoid loading `[refactor-nrepl "2.2.0"]` in my `.lein/profiles.clj` then everything works perfectly again.

Interestingly, the issue is only triggered when both `refactor-nrepl` is declared and _after_ CIDER in Emacs has connected. The connection from CIDER causes `refactor-nrepl` to load something that causes the initial compilation of my code graph (via `(dev)`, but no matter), to fail. Oddly, it fails in a number of different ways - it seems non-deterministic.

There's always the chance this issue will trip someone else up, and search engines are fairly good and picking up GitHub issues, so perhaps someone may in future throw some more light on this. In the meantime, I'll link this issue to a new one in https://github.com/clojure-emacs/refactor-nrepl
